numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -373,29 +374,24 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 gentype_str(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
-
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    static PyObject *fallback = NULL;
+    npy_cache_import("numpy.core.arrayprint", "_scalar_fallback", &fallback);
+    if (fallback == NULL) {
+        return NULL;
     }
-    return ret;
+    return PyObject_CallFunction(fallback, "O", self);
```

Comment:
> we will get infinite recursions

Hopefully they should now be bounded - that was why wanted that other commit merged

That doesn't seem sane as a formatter though anyway - "Changing the `__str__` for arrays to call `str(self)`" is obviously very wrong